lib/widgets/clipboard.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// Copies [data] to the clipboard and shows a popup on success.
+///
+/// Must have a [Scaffold] ancestor.
+///
+/// On newer Android the popup is defined and shown by the platform. On older
+/// Android and on iOS, shows a [Snackbar] with [successText].
+///
+/// In English, the [successText] should be short, should start with a capital
+/// letter, and should have no ending punctuation: "{noun} copied".
+void copyWithPopup({
+  required BuildContext context,
+  required ClipboardData data,
+  required Text successText // TODO(i18n)
+}) async {
+  await Clipboard.setData(data);
+  final deviceInfo = await DeviceInfoPlugin().deviceInfo;
+
+  // Early return on !mounted would be better, but:
+  //   https://github.com/dart-lang/linter/issues/4007
+  if (context.mounted) {
+    final shouldShowSnackbar =
+      (deviceInfo is IosDeviceInfo)
+
+        // Android 13+ shows its own popup on copying to the clipboard,
+        // so we suppress ours, following the advice at:
+        //   https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications
+        // TODO(android-sdk-32) Remove and simplify dartdoc
+        || (deviceInfo is AndroidDeviceInfo && deviceInfo.version.sdkInt <= 32);
```

Comment:
This is a bit hard to read because the conditions are separated so widely from each other by the comment, it looks after the "IosDeviceInfo" line as if that's the whole condition.

One solution could be to move the comment above the whole expression.

Or, here's a version using fancy Dart 3 pattern-matching:
```suggestion
    final bool shouldShowSnackbar;
    switch (deviceInfo) {
      case AndroidDeviceInfo(:var version):
        // Android 13+ shows its own popup on copying to the clipboard,
        // so we suppress ours, following the advice at:
        //   https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications
        // TODO(android-sdk-33): Simplify this and dartdoc
        shouldShowSnackbar = version.sdkInt <= 32;
      default:
        shouldShowSnackbar = true;
    };
```

Reply:
Also reflected in that version: if SDK 32 is the last version with the old behavior, then SDK 33 is the one where this TODO becomes actionable when we bump the minimum that far, which means that's the one we want to mention in the comment.